### PR TITLE
Parse bug: negations for inequality constraints are ignored

### PR DESCRIPTION
**Pull request by SimpleXiaohu**
You seem to forget to use the `isPositive` flag when transferring `Inequality` to princess formula. The following instance is `sat`, but Catra answers `unsat`:
```sh
counter int R3, Int1, R1, R2, all_2_0;
synchronised {
automaton aut_X0 {
	init s3;
	s0 -> s1 [97, 97] {R2 += 1};
	s1 -> s1 [97, 97] {R2 += 1};
	s2 -> s2 [48, 48] {R1 += 1, R2 += 1};
	s3 -> s2 [48, 48] {R1 += 1, R2 += 1};
	s3 -> s4 [97, 97] {R2 += 1};
	s4 -> s0 [97, 97] {R2 += 1};
	accepting s0, s1, s2;
};
automaton aut_X1 {
	init s0;
	s0 -> s0 [0, 65535] {R3 += 1};
	accepting s0;
};
};
constraint (R3 + -1*all_2_0 = 0 && all_2_0 + -21 >= 0 && ! (-1*R2 >= 0 && ! (-1*R1 + 1 >= 0 && R1 + -1 >= 0)));

```

My pull request fixes this bug

### CODE REVIEW

**Comment by SimpleXiaohu** [comment]
@amandasystems

File: src/main/scala/uuverifiers/catra/NUXMVBackend.scala [hunk 1]
```diff
@@ -223,6 +223,8 @@ class NUXMVInstance(arguments: CommandLineOptions, instance: Instance)
       }
       out.close()
 
+      printNUXMVModule()
```
Comment (amandasystems) [inline src/main/scala/uuverifiers/catra/NUXMVBackend.scala]: Could you please remove this?

**Comment by amandasystems** [comment]
Nice, thanks! Could you remove the print statement from the nuxmv module and then I’ll merge!

**Comment by SimpleXiaohu** [comment]
No problem. I removed the print statement. Furthermore, I  lightly changed the nuxmv encoding to model it in one step ( two steps in the previous encoding)

**Comment by amandasystems** [comment]
I'm getting some weird results when I merge the whole commit; I'm trying to figure out why that is, might be spurious.

**Comment by amandasystems** [comment]
I can confirm: the nuXmv backend behaves erratically when merging your changes to it. I will cherry-pick the updates to the input file parser and merge only those!

Reproduction:
```
$ ./bin/catra solve-satisfy --timeout 30000 --cross-validate basket
```

starts yielding errors.

**Pull request closed without merging**